### PR TITLE
feat-no media types in paths

### PR DESCRIPTION
**Pull request by savage-alex**
New rule to address issue https://github.com/apisyouwonthate/style-guide/issues/24
Tests added
Tested with spectral CLI v6.5.0 and VS code extention v1.0

### CODE REVIEW

File: src/ruleset.ts [hunk 1]
```diff
@@ -278,5 +278,20 @@ export default {
...
+      message:
+        "Paths must not include file extensions such as .json, .xml, .html and .txt. Use the OpenAPI `content` keyword to tell consumers which Media Types are available.",
+        given: "$.paths[*]~",
+        then: {
+        function: pattern,
+        functionOptions: {
+          notMatch: ".(\\.json|\\.xml|\\.html|\\.txt)$",
+        },
+      },
+      formats: [oas3],
```
Comment (philsturgeon) [inline src/ruleset.ts:293]: Any reason this is only being applied to oas3? I think this will work on OAS2 as well.

Comment (savage-alex) [reply src/ruleset.ts:293]: Oversight is the only reason. Let me give it some tests and if it's fine, add it there too.

**Comment by savage-alex** [comment]
@philsturgeon Sadly unable to merge this as the Tests are stuck.

File: src/ruleset.ts (hunk 1 above)
Comment (philsturgeon) [reply src/ruleset.ts:293]: @savage-alex can you see if this works on oas2 now we've fixed #32.

**Comment by philsturgeon** [comment]
@savage-alex tests are no longer stuck.

File: src/ruleset.ts (hunk 1 above)
Comment (philsturgeon) [reply src/ruleset.ts:294]:
```suggestion
```

File: src/ruleset.ts [hunk 2]
```diff
@@ -279,6 +279,21 @@ export default {
...
+    // Author: Advanced API & Integrations Team (https://www.oneadvanced.com/)
+    "no-file-extensions-in-paths": {
+      description: "Paths must not include file extensions such as .json, .xml, .html and .txt",
+      message:
+        "Paths must not include file extensions such as .json, .xml, .html and .txt. Use the OpenAPI `content` keyword to tell consumers which Media Types are available.",
+        given: "$.paths[*]~",
+        then: {
+        function: pattern,
+        functionOptions: {
+          notMatch: ".(\\.json|\\.xml|\\.html|\\.txt)$",
```
Comment (philsturgeon) [inline src/ruleset.ts:291]:
```suggestion
          notMatch: "\.(json|xml|html|txt)$",
```

**Comment by savage-alex** [comment]
@philsturgeon OAS2 tested fine and tests added.

**Review by philsturgeon — APPROVED** [review]

**Pull request merged**